packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -1769,6 +1871,16 @@ class _InputDecoratorState extends State<InputDecorator> with TickerProviderStat
     return lightEnabled;
   }
 
+  Color _getFocusColor(ThemeData themeData) {
+    if (decoration.filled != true) // filled == null same as filled == false
+      return Colors.transparent;
+    return decoration.focusColor ?? themeData.inputDecorationTheme?.focusColor ?? themeData.focusColor;
+  }
+
+  Color _getHoverColor(ThemeData themeData) {
```

Comment:
Why does `_getFocusColor()` handle the `decoration.filled` case, but not `_getHoverColor()`?

Reply:
Because it is wrong. Fixed.